tests/boards/nrf/coresight_stm/CMakeLists.txt

```diff
@@ -0,0 +1,17 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+cmake_minimum_required(VERSION 3.20.0)
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+
+if(NOT (CONFIG_BOARD_NRF54H20DK_NRF54H20_CPUAPP))
+  message(FATAL_ERROR "${BOARD}${BOARD_QUALIFIERS} is not supported for this sample")
+endif()
```

Comment:
This is already specified in `testcase.yaml` so seems redundant.

Reply:
testcase.yaml is parsed by Twister only. This check is valid for both Twister and manual `west build`.